Core/Src/Systems/External/BMS.c

```diff
@@ -0,0 +1,118 @@
+#include "../../../Inc/Systems/External/BMS.h"
+
+#include <stdio.h>  // For printf
+#include "../../../Inc/Systems/External/Can/Can.h"
+
+void initBms(Bms* bms, int hz, const char* dbcFn) {
+    if (bms == NULL) return;
+
+    initExternalSystem(&bms->extSystem, "Bms", hz, 0);
+
+    bms->packVoltage = 0.0f;
+    bms->packCurrent = 0.0f;
+    bms->stateOfCharge = 0.0f;
+    bms->cellVoltageMin = 0.0f;
+    bms->cellVoltageMax = 0.0f;
+    bms->cellTemperatureMin = 0.0f;
+    bms->cellTemperatureMax = 0.0f;
+    bms->totalPackCapacity = 0.0f;
+    bms->remainingPackCapacity = 0.0f;
+    bms->packHealth = 0.0f;
+    bms->chargeStatus = IDLE;
+    bms->extSystem.system.updateable.update = updateBms;
+
+    DBC dbc;
+    if(!parseDbcFile(&dbc, dbcFn)) {
+        printf("Error: Couldn't parse dbc.\n");
+        return;
+    }
+    bms->dbc = &dbc;
```

Comment:
The current implementation creates a `DBC` variable on the stack and assigns its address to `bms->dbc`. 
This could be risky since:
1. The DBC struct only exists within the scope of `initBms`
2. After `initBms` returns, `bms->dbc` points to stack memory that is no longer valid

Suggested fix:
Either:
1. Allocate DBC on heap using `malloc`:
   ```c
   bms->dbc = malloc(sizeof(DBC));
   if(!parseDbcFile(bms->dbc, dbcFn)) {
       free(bms->dbc);
       return;
   }
   ```
2. Or store DBC directly in BMS struct:
    ```c
    typedef struct {
        DBC dbc;  // Not a pointer
        // ...
    } Bms;
    ```